windows/devices/uwb/UwbSession.cxx

```diff
@@ -14,12 +16,152 @@ UwbSession::UwbSession(std::weak_ptr<uwb::UwbSessionEventCallbacks> callbacks, w
     m_handleDriver(std::move(handleDriver))
 {}
 
+template <class T>
+std::unique_ptr<UWB_APP_CONFIG_PARAM>
+GetUWB_APP_CONFIG_PARAM(const T &value)
+{
+    auto pdata = std::make_unique<uint8_t>(FIELD_OFFSET(UWB_APP_CONFIG_PARAM, paramValue[sizeof T]));
+}
```

Comment:
This needs to use the array overload (`std::make_unique<uint8_t[]>`). Also, `std::unique_ptr<uint8_t[]>` is incompatible with `std::unique_ptr<UWB_APP_CONFIG_PARAM>`; attempting to force it with `static_cast` will point out that one doesn't derive from the other. 

There's no way to allocate a flex-sized `UWB_APP_CONFIG_PARAM` that can be used with `std::unique_ptr`, because the deleter will always be bound to the contained type. 

There needs to be two stages: one is allocating the byte buffer, you've got that already, the second is interpreting the buffer as the intended type `T`. You could do something like this:

```suggestion
template <class T>
std::tuple<std::unique_ptr<uint8_t[]>, UWB_APP_CONFIG_PARAM*, T*>
GetUWB_APP_CONFIG_PARAM(const T &value)
{
    auto appConfigSize = offsetof(UWB_APP_CONFIG_PARAM, paramValue[sizeof(T)]);
    auto appConfigBuffer = std::make_unique<uint8_t[]>(appConfigSize);
    auto appConfigParam = reinterpret_cast<UWB_APP_CONFIG_PARAM*>(appConfigBuffer.get());
    auto appConfigValue = reinterpret_cast<T&>(&appConfigParam->paramValue[0]);
    return { appConfigBuffer, appConfigParam, appConfigValue };
}
```

This gets messy though, since the ownership of the tuple contents is unclear, and they could get passed around individually and easily lose their meaning/binding. An adapter class could manage all of that. Eg.

```
template <class T>
class UwbAppConfigAdapter
{
    UwbAppConfigAdapter(const T& value) : 
        m_buffer(std::make_unique<uint8_t[]>(offsetof(UWB_APP_CONFIG_PARAM), paramValue[sizeof(T)])),
        m_parameter(*reinterpret_cast<UWB_APP_CONFIG_PARAM*>(m_buffer.get()))
        m_value(*reinterpret_cast<T*>(&m_parameter.paramValue[0]))
    {
        m_value = value; // set value contents
    }
    
    T& Value() { return m_value; }
    UWB_APP_CONFIG_PARAM& Parameter() { m_parameter; }
    
private:
    T& m_value;
    UWB_APP_CONFIG_PARAM &m_parameter;
    std::unique_ptr<uint8_t[]> m_buffer;
};
```
Which then gets used as:

```
UwbAppConfigAdapter<uwb::protocol::fira::DeviceRole> uwbAppConfigDeviceRole { uwb::protocol::fira::DeviceRole::Initiator };
auto role = uwbAppConfigDeviceRole.Value();
auto appConfigParam = uwbAppConfigDeviceRole.Parameter(); // pass me to DeviceIoControl()
```

Reply:
#23